stores/utxo/src/main/java/com/bloxbean/cardano/yaci/store/utxo/storage/impl/UtxoStorageReaderImpl.java

```diff
@@ -54,6 +55,27 @@ public List<AddressUtxo> findUtxoByAddress(@NonNull String address, int page, in
                 .toList();
     }
 
+    @Override
+    public List<AddressUtxo> findUtxosByAsset(String unit, int page, int count, Order order) {
+        Pageable pageable = PageRequest.of(page, count)
+                .withSort(order.equals(Order.desc) ? Sort.Direction.DESC : Sort.Direction.ASC, "slot", "txHash", "outputIndex");
+
+        var query = dsl
+                .select(ADDRESS_UTXO.fields())
+                .from(ADDRESS_UTXO)
+                .leftJoin(TX_INPUT)
+                .using(field(ADDRESS_UTXO.TX_HASH), field(ADDRESS_UTXO.OUTPUT_INDEX))
+                .where(field(ADDRESS_UTXO.AMOUNTS).cast(String.class).equalIgnoreCase(unit))
```

Comment:
@edridudi This may not work as AMOUNTS is a json array. So exact matching will not work.
Let's keep it as "contains" for now.

We will create another issue to fix this. We need to use json functions from JOOQ https://www.jooq.org/doc/latest/manual/sql-building/column-expressions/json-functions/json-attribute-access/